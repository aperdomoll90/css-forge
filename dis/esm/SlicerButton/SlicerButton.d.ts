import React from 'react';
import './styles.css';
import '../utils/GlobalStyles.css';
import { SlicerButtonPropsTypes } from './SlicerButton.types';
export declare const SlicerButton: React.FC<SlicerButtonPropsTypes>;
