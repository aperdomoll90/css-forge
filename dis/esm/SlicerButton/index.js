export { SlicerButton } from './SlicerButton';
export * from './SlicerButton.types';
//# sourceMappingURL=index.js.map