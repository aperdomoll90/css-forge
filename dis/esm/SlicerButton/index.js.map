{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../src/SlicerButton/index.tsx"],"names":[],"mappings":"AAAA,OAAO,EAAE,YAAY,EAAE,MAAM,gBAAgB,CAAA;AAC7C,cAAc,sBAAsB,CAAA"}