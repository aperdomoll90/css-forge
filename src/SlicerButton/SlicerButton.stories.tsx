import { Meta, StoryFn } from '@storybook/react'
import { SlicerButton } from './SlicerButton'
import StorybookContainer from '../StorybookContainer'
import { SlicerButtonPropsTypes } from './SlicerButton.types'

export default {
  title: 'Buttons',
  component: SlicerButton,
  parameters: {
    jest: ['SlicerButton.test.tsx'],
  },
  argTypes: {
    label: { options: ['Button', 'Contact me', 'Home'], control: { type: 'radio' } },
    fontSize: { options: ['1rem', '3rem', '5rem'], control: { type: 'radio' } },
    color: { options: ['#d93654', '#000', 'yellow', 'purple', ''], control: { type: 'radio' } },
    colorHover: { options: ['#d93654', '#000', 'yellow', 'purple', ''], control: { type: 'radio' } },
  },
} as Meta

const SlicerButtonTemplate: StoryFn<SlicerButtonPropsTypes> = args => (
  <StorybookContainer>
    <div className='story-button-container-centered'>
      <SlicerButton {...args} />
    </div>
  </StorybookContainer>
)

export const SlicerButtonComponent = SlicerButtonTemplate.bind({})
SlicerButtonComponent.args = {
  label: 'Button',
  linkUrl: '#testAddress',
  color: 'red',
  fontSize: '2rem',
  colorHover: 'yellow',
}
