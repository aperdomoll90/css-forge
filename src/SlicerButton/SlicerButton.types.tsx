export interface SlicerButtonPropsTypes {
  label: string
  linkUrl: string
  color: string
  fontSize: string
  colorHover: string
}
