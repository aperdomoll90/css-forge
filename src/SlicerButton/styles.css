.slicerButton-wrapper {
  position: relative;
}

h2 {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  margin: 0;
  padding: 0;
  font-size: var(--fontSize);
  font-weight: 900;
  color: transparent;
  text-transform: uppercase;
  font-family: 'Roboto', sans-serif;
}
h2 span:nth-child(1) {
  position: absolute;
  top: 0;
  left: 0;
  color: var(--color);
  transition: 0.5s;
  clip-path: polygon(0 0, 100% 0, 100% 50%, 0 50%);
  overflow: hidden;
}
h2:hover span:nth-child(1) {
  color: var(--colorHover);
  transform: translateX(-10px);
}

h2 span:nth-child(2) {
  position: absolute;
  top: -0.01;
  left: 0;
  color: var(--color);
  transition: 0.5s;
  clip-path: polygon(0 49.8%, 100% 49.8%, 100% 100%, 0 100%);
  overflow: hidden;
}
h2:hover span:nth-child(2) {
  color: var(--colorHover);
  transform: translateX(10px);
}

/* @media only screen and (max-width: 1625px) {
  h2 {
    font-size: var(--fontSize)
  }
}

@media only screen and (max-width: 767px) {
  h2 {
    font-size: var(--fontSize);

  }
  h2 {
    font-size: var(--fontSize);

  }
} */
