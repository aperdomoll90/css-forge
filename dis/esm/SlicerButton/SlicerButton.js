var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { jsxs as _jsxs, jsx as _jsx } from "react/jsx-runtime";
import './styles.css';
import '../utils/GlobalStyles.css';
export var SlicerButton = function (_a) {
    var label = _a.label, linkUrl = _a.linkUrl, color = _a.color, fontSize = _a.fontSize, colorHover = _a.colorHover;
    var defaultColor = '#d93654';
    var defaultColorHover = '#f2f2f2';
    return (_jsx("a", __assign({ href: linkUrl }, { children: _jsx("div", __assign({ className: 'slicerButton-wrapper', style: { '--fontSize': fontSize ? fontSize : '1rem', '--color': color ? color : defaultColor, '--colorHover': colorHover ? colorHover : defaultColorHover, } }, { children: _jsxs("h2", { children: [label, _jsxs("span", { children: [" ", label] }), _jsxs("span", { children: [" ", label] })] }) })) })));
};
//# sourceMappingURL=SlicerButton.js.map