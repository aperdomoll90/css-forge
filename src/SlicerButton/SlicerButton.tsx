import React from 'react'
import './styles.css'
import '../utils/GlobalStyles.css'
import { SlicerButtonPropsTypes } from './SlicerButton.types'

export const SlicerButton: React.FC<SlicerButtonPropsTypes> = ({ label, linkUrl, color, fontSize, colorHover }) => {
  const defaultColor = '#d93654'
  const defaultColorHover = '#f2f2f2'
  return (
    <a href={linkUrl}>
      <div className='slicerButton-wrapper' style={{ '--fontSize': fontSize ? fontSize : '1rem','--color': color ? color : defaultColor, '--colorHover': colorHover ? colorHover : defaultColorHover, } as React.CSSProperties}>
        <h2>
          {label}
          <span> {label}</span>
          <span> {label}</span>
        </h2>
      </div>
    </a>
  )
}
