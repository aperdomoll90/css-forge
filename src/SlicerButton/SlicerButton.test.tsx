import { render } from '@testing-library/react'
import { SlicerButton } from './SlicerButton'
import { SlicerButtonPropsTypes } from './SlicerButton.types'

describe('SlicerButton component', () => {
  const props: SlicerButtonPropsTypes = {
    label: 'Button',
    linkUrl: '#testAddress',
    color: 'red',
    fontSize: '2rem',
    colorHover: 'yellow',
  }

  const renderComponent = () => render(<SlicerButton {...props} />)
  it('should match snapshots', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
