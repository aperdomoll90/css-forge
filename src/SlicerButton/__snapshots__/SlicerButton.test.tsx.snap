// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlicerButton component should match snapshots 1`] = `
<div>
  <a
    href="#testAddress"
  >
    <div
      class="slicerButton-wrapper"
      style="--fontSize: 2rem; --color: red; --colorHover: yellow;"
    >
      <h2>
        Button
        <span>
           
          Button
        </span>
        <span>
           
          Button
        </span>
      </h2>
    </div>
  </a>
</div>
`;
